compiler-plugin/src/main/kotlin/arrow/meta/plugins/proofs/phases/quotes/SerializeRefinedPredicates.kt

```diff
@@ -1,19 +1,28 @@
 package arrow.meta.plugins.proofs.phases.quotes
 
+import arrow.meta.phases.CompilerContext
 import arrow.meta.phases.analysis.ElementScope
+import arrow.meta.phases.evaluateDependsOn
 import arrow.meta.quotes.Transform
 import arrow.meta.quotes.classorobject.ObjectDeclaration
 import org.jetbrains.kotlin.psi.KtObjectDeclaration
 
-internal fun ObjectDeclaration.objectWithSerializedRefinement(elementScope: ElementScope): Transform<KtObjectDeclaration> =
-  elementScope.run {
+internal fun ObjectDeclaration.objectWithSerializedRefinement(elementScope: ElementScope, ctx: CompilerContext): Transform<KtObjectDeclaration> =
+  ctx.evaluateDependsOn(
```

Comment:
Is there a way to encapsulate this reoccurring pattern in the internals so it does not have to be repeated in each quote? Seems boilerplate except `evaluatesRefinementExpression(elementScope)`